src/integration-tests/src/observability/cloud_trace.rs

```diff
@@ -0,0 +1,164 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use auth::credentials::{CacheableResource, Credentials, EntityTag};
+use http::Extensions;
+use std::time::Duration;
+
+/// A client for the Google Cloud Trace API (v1 REST).
+///
+/// It uses the v1 rest API for simplicity since there is no generated client that can get traces.
+pub struct CloudTraceClient {
+    project_id: String,
+    http_client: reqwest::Client,
+    credentials: Credentials,
+    endpoint: String,
+}
+
+/// Builder for `CloudTraceClient`.
+pub struct CloudTraceClientBuilder {
+    project_id: String,
+    credentials: Option<Credentials>,
+    endpoint: String,
+}
+
+impl CloudTraceClientBuilder {
+    pub fn new(project_id: impl Into<String>) -> Self {
+        Self {
+            project_id: project_id.into(),
+            credentials: None,
+            endpoint: "https://cloudtrace.googleapis.com".to_string(),
+        }
+    }
+
+    /// Sets the credentials used for authentication.
+    /// If not provided, Application Default Credentials (ADC) will be loaded.
+    pub fn with_credentials(mut self, credentials: Credentials) -> Self {
+        self.credentials = Some(credentials);
+        self
+    }
+
+    /// Sets the endpoint for the Cloud Trace API.
+    /// Defaults to `https://cloudtrace.googleapis.com`.
+    pub fn with_endpoint(mut self, endpoint: impl Into<String>) -> Self {
+        self.endpoint = endpoint.into();
+        self
+    }
+
+    pub async fn build(self) -> anyhow::Result<CloudTraceClient> {
+        let credentials = match self.credentials {
+            Some(c) => c,
+            None => auth::credentials::Builder::default().build()?,
+        };
+
+        Ok(CloudTraceClient {
+            project_id: self.project_id,
+            http_client: reqwest::Client::new(),
+            credentials,
+            endpoint: self.endpoint,
+        })
+    }
+}
+
+impl CloudTraceClient {
+    /// Creates a builder for `CloudTraceClient`.
+    pub fn builder(project_id: impl Into<String>) -> CloudTraceClientBuilder {
+        CloudTraceClientBuilder::new(project_id)
+    }
+
+    /// Polls the Cloud Trace API for a specific trace ID.
+    ///
+    /// This method will retry up to `max_retries` times, waiting `interval`
+    /// between attempts.
+    pub async fn get_trace(
+        &self,
+        trace_id: &str,
+        max_retries: usize,
+        interval: Duration,
+    ) -> anyhow::Result<serde_json::Value> {
+        let url = format!(
+            "{}/v1/projects/{}/traces/{}",
+            self.endpoint, self.project_id, trace_id
+        );
+
+        let mut cached_token: Option<String> = None;
+        let mut last_etag: Option<EntityTag> = None;
+
+        for i in 1..=max_retries {
+            let token = self.get_token(&mut cached_token, &mut last_etag).await?;
+
+            let resp = self
+                .http_client
+                .get(&url)
```

Comment:
comment: I remember this sucking.

As you noticed, it takes a while for the trace to become available, and the RPC has a quota on it, so we cannot just retry willy-nilly.

This is the loop I ended up with in C++, which starts with an initial sleep of 10s (yuck!):

https://github.com/googleapis/google-cloud-cpp/blob/f928005d4b02cec2d1896fe3575ec44170a0042d/google/cloud/opentelemetry/integration_tests/trace_exporter_integration_test.cc#L99-L107